lib/withOnyx.js

```diff
@@ -244,7 +256,14 @@ export default function (mapOnyxToState, shouldDelayUpdates = false) {
              *  component
              */
             connectMappingToOnyx(mapping, statePropertyName) {
-                const key = Str.result(mapping.key, this.props);
+                const key = Str.result(mapping.key, {...this.props, ...getOnyxDataFromState(this.state, mapOnyxToState)});
+
+                // Remember the previous key so that if it ever changes, the component will reconnect to Onyx
+                // in componentDidUpdate
+                if (statePropertyName !== 'initialValue' && mapOnyxToState[statePropertyName]) {
+                    // eslint-disable-next-line no-param-reassign
+                    mapOnyxToState[statePropertyName].previousKey = key;
+                }
```

Comment:
This is another small change from the previous PR. When `initialValue` was integrated, it lead to a situation where `mapOnyxToState.initialValue: false`, which caused `mapOnyxToState[statePropertyName].previousKey` to throw an error (because you cannot set `.previousKey` on the value `false`).